mlflow/gateway/providers/openai.py

```diff
@@ -0,0 +1,150 @@
+import openai
+from fastapi.encoders import jsonable_encoder
+
+from .base import BaseProvider
+from ..schemas import chat, completions, embeddings
+
+
+class OpenAIProvider(BaseProvider):
+    NAME = "openai"
+    SUPPORTED_ROUTES = ("chat", "completions", "embeddings")
+
+    async def chat(self, payload: chat.RequestPayload) -> chat.ResponsePayload:
+        resp = await openai.ChatCompletion.acreate(
```

Comment:
Were we going to use their REST endpoints instead of their SDK?

Reply:
I remember we discussed this, but forgot our conclusion.

Reply:
I think we should use REST so we can standardize retry logic

Reply:
Makes sense, I'll update the code.

Reply:
Updated!

Reply:
Testing now

Reply:
Tests need to be updated

Reply:
Should be working now.